### ISSUE TITLE
susbtitution bug in SP?

### ISSUE THREAD

**Issue by bqpd** [hoburg/gpkit#846]
Re #828 and #833 only happening with SPs, the fix works but may be incomplete.

**Comment by whoburg** [comment]
(Did you mean something other than #828? That seems an odd issue for this one to refer to.)

**Comment by bqpd** [comment]
Hmm maybe I meant #832?

**Comment by whoburg** [comment]
Adding some detail here: this issue refers to the fact that the following problem from #832:

```python


import gpkit
from gpkit.constraints.tight import TightConstraintSet as TCS

gpkit.SIGNOMIALS_ENABLED = True

x = gpkit.Variable("x")
y = gpkit.Variable("y")
ymax = gpkit.Variable("y_{max}", 0.1)


m = gpkit.Model(x, [x >= 0.5, TCS([x >= 1-y, y <= ymax])])
m.substitutions[ymax] = 0.2
m.localsolve()

m = gpkit.Model(x, [x >= 0.5, TCS([x >= 1-y, y <= ymax])])
m.localsolve() # should end with a y of 0.1, instead has 0.2
```

... occurred only for SPs, and not for a similar GP, e.g.
```python
x = Variable("x")
y = Variable("y")
ymin = Variable("y_{min}", 0.1)
m = Model(x, ConstraintSet([x >= y, y >= ymin]))
```

Although the original bug was patched by #833, goal of this ticket is to understand why it occurred for SPs only and not GPs. It can be reproduced at commit 142c21711914b08c89ca11a136e1e94f04b0f991.

**Comment by bqpd** [comment]
Seems to be fixed on current master
